go/consensus/cometbft/seed/seed.go

```diff
@@ -125,6 +126,21 @@ func (s *Service) GetAddresses() ([]node.ConsensusAddress, error) {
 	return []node.ConsensusAddress{addr}, nil
 }
 
+func initSeedDataDir(dataDir string) (string, error) {
+	seedDataDir := filepath.Join(dataDir, "seed")
+
+	// Check if there is a legacy data directory and migrate it so we can get the old address book.
+	legacyDataDir := filepath.Join(dataDir, "tendermint-seed")
```

Comment:
After the upgrade, this code will become obsolete. To ensure its removal is not forgotten, I prefer preparing a draft pull request in advance for these cases.